### ISSUE TITLE
Payment approved not being sent

### ISSUE THREAD

**Issue by hknobbe** [mollie/PrestaShop#180]
Hi all,

Just found a strange new glitch/bug.
After a payment has been made (ordersAPI), the state of the order is not being changed. I then have to manually forward the order. Strange enough, after some time it still comes trough, but that seems a little bit at random.

Backoffice Mollie - ORDER:
![Screenshot_9](https://user-images.githubusercontent.com/39634152/85382286-16e69a00-b53f-11ea-8ba3-de54f0fb9956.png)

Backoffice Mollie - PAYMENT:
![Screenshot_10](https://user-images.githubusercontent.com/39634152/85382390-3251a500-b53f-11ea-8847-25c7116263d4.png)

Prestashop - ORDER:
![Screenshot_11](https://user-images.githubusercontent.com/39634152/85382470-4b5a5600-b53f-11ea-9c14-1c06da59a3e1.png)

![Screenshot_12](https://user-images.githubusercontent.com/39634152/85382792-996f5980-b53f-11ea-8d2c-c79cff0aaff9.png)

Prestashop - Mollie settings:
![Screenshot_13](https://user-images.githubusercontent.com/39634152/85383023-cf144280-b53f-11ea-8aa9-a34738fa1711.png)

Error log:
```
[Tue Jun 23 09:03:02.993260 2020] [proxy_fcgi:error] [pid 6725:tid 140105465399040] [client 172.69.63.147:31740] AH01071: Got error 'PHP message: PHP Warning:  html_entity_decode() expects parameter 1 to be string, object given in /var/www/html/classes/Tools.php on line 1795\n'
[Tue Jun 23 09:03:56.996488 2020] [proxy_fcgi:error] [pid 6726:tid 140105540933376] [client 141.101.104.167:60908] AH01071: Got error 'PHP message: PHP Warning:  html_entity_decode() expects parameter 1 to be string, object given in /var/www/html/classes/Tools.php on line 1795\n'
[Tue Jun 23 09:04:17.442115 2020] [proxy_fcgi:error] [pid 6725:tid 140105582896896] [client 162.158.158.21:27578] AH01071: Got error 'PHP message: PHP Warning:  html_entity_decode() expects parameter 1 to be string, object given in /var/www/html/classes/Tools.php on line 1795\n'

```

```
Prestashop: 1.7.6.5
PHP-version: 7.2.31-1+ubuntu18.04.1+deb.sury.org+1
MySQL version: 10.1.44-MariaDB-0ubuntu0.18.04.1
Mollie: 4.0.5
```

Let me know if you need anything more.

Thanks and best regards,
Henk-Jan

**Comment by hknobbe** [comment]
Small update after some more testing;

It looks like that people don't always wait for the confirmation page after (for example) an iDeal payment. I think thats why the order stays on waiting for mollie payment.

Maybe there could be a way arround this in a future version?

Thanks!

**Comment by margud** [comment]
Issue was resolved in [4.0.6](https://github.com/mollie/PrestaShop/releases/tag/4.0.6) release.
